client/trino-client/src/main/java/io/trino/client/spooling/EncodedQueryData.java

```diff
@@ -112,6 +112,12 @@ public Builder withSegment(Segment segment)
             return this;
         }
 
+        public Builder withSegments(List<Segment> segment)
```

Comment:
nit. It can be `Segment... segments`

Even if you keep `List<>` rename param to `segments`